### ISSUE TITLE
Bug in tag propagation

### ISSUE THREAD

**Issue by unageek** [unageek/boole#1]
Inputs:

star_a.off

```
OFF
14 12 0
-1 -1 -1
1 -1 -1
-1 1 -1
1 1 -1
0 -1 0
0 1 0
-1 -1 1
1 -1 1
-1 1 1
1 1 1
-2 -1 0
2 -1 0
-2 1 0
2 1 0
3  0 2 5
3  0 5 4
3  3 1 4
3  3 4 5
3  5 4 6
3  5 6 8
3  4 5 9
3  4 9 7
3  10 4 5
3  10 5 12
3  4 11 13
3  4 13 5
```

star_b.off

```
OFF
4 2 0
-2 -1 0
2 -1 0
-2 1 0
2 1 0
3  0 1 3
3  0 3 2
```

**Comment by unageek** [comment]
In the current implementation, tag propagation can cross shared edges, but it must not.

**Comment by unageek** [comment]
Fixed in c0a28792fe968d0c559af2c02dc183f57d0504fa.
